src/openrct2/ride/TrackDesign.cpp

```diff
@@ -1960,13 +1961,17 @@ static money32 place_track_design(int16_t x, int16_t y, int16_t z, uint8_t flags
 
     if (entryIndex != 0xFF)
     {
-        game_do_command(0, flags | (2 << 8), 0, ride->id | (entryIndex << 8), GAME_COMMAND_SET_RIDE_VEHICLES, 0, 0);
+        auto colour = ride_get_unused_preset_vehicle_colour(entryIndex);
+        auto rideSetVehicleAction = RideSetVehicleAction(ride->id, RideSetVehicleType::RideEntry, entryIndex, colour);
+        GameActions::ExecuteNested(&rideSetVehicleAction);
```

Comment:
Missing set flags!

Reply:
Flags it seems are not used in this function but there should have been an QueryNested if.

Reply:
Well the game action base class uses some flags to determine the game action flags, but yeah since its nested it doesn't matter.